extensions/blocks/podcast-player/view.js

```diff
@@ -58,26 +59,224 @@ if ( window.jetpackPodcastPlayers !== undefined ) {
+
+	if ( episodeLinkEl ) {
+		// Prevent handling clicks if a modifier is in use.
+		if ( e.shiftKey || e.metaKey || e.altKey ) {
+			return;
+		}
+
+		e.preventDefault();
+		handleEpisodeLinkClick( episodeLinkEl );
+	}
+}
+
+// Add document event listeners
+document.addEventListener( 'click', handleEpisodeLinkEvent );
+document.addEventListener( 'keydown', handleEpisodeLinkKeydown );
+
+function handleEpisodeLinkClick( episodeLinkEl ) {
+	// Get clicked episode element
+	const episodeEl = episodeLinkEl.closest( '.podcast-player__episode' );
+	if ( ! episodeEl ) {
+		// Append the error to closest parent if episode element is not present.
+		return handleError( episodeLinkEl.closest( '*' ) );
+	}
+
+	// Get clicked episode audio URL
+	const audioUrl = episodeLinkEl.getAttribute( 'data-jetpack-podcast-audio' );
+	if ( ! audioUrl ) {
+		return handleError( episodeEl );
+	}
+
+	// Get episode's parent block element
+	const blockEl = episodeEl.closest( '.wp-block-jetpack-podcast-player' );
+	if ( ! blockEl ) {
+		return handleError( episodeEl );
+	}
+
+	// Get player instance by block id
+	const player = playerInstances[ blockEl.id ];
+	if ( ! player ) {
+		return handleError( episodeEl );
+	}
+
+	// Get currently active episode element
+	const activeEpisodeEl = document.querySelector( '.podcast-player__episode.is-active' );
+	if ( activeEpisodeEl ) {
+		if ( activeEpisodeEl.isSameNode( episodeEl ) ) {
+			if ( player.audio.paused ) {
+				player.audio.play();
+				handlePlay( activeEpisodeEl );
+			} else {
+				player.audio.pause();
+				handlePause( activeEpisodeEl );
+			}
+			return;
 		}
+		// Make episode inactive
+		activeEpisodeEl.classList.remove( 'is-active' );
+		activeEpisodeEl
+			.querySelector( '[aria-pressed="true"]' )
+			.setAttribute( 'aria-pressed', 'false' );
+
+		handlePause( activeEpisodeEl );
+	}
+
+	player.audio.src = audioUrl;
+
+	player.audio
+		.play()
+		.then( () => handlePlay( episodeEl ) )
+		.catch( () => handleError( episodeEl ) );
+}
+
+function renderEpisodeError( episodeEl ) {
+	const parentBlockEl = episodeEl.closest( '.wp-block-jetpack-podcast-player' );
+
+	// Don't render if already rendered
+	if ( parentBlockEl.querySelector( '.podcast-player__episode-error' ) ) {
+		return;
+	}
+
+	const episodeLinkEl = episodeEl.querySelector( '.podcast-player__episode-link' );
+
+	const linkEl = document.createElement( 'a' );
+	linkEl.rel = 'noopener noreferrer nofollow';
+	linkEl.target = '_blank';
+	linkEl.href = episodeLinkEl.href;
+	linkEl.innerText = __( 'Open in a new tab', 'jetpack' );
+
+	const spanEl = document.createElement( 'span' );
+	spanEl.appendChild( new Text( '(' ) );
+	spanEl.appendChild( linkEl );
+	spanEl.appendChild( new Text( ')' ) );
+
+	const errorEl = document.createElement( 'div' );
+	errorEl.classList.add( 'podcast-player__episode-error' );
+	errorEl.appendChild( new Text( __( 'Episode unavailable', 'jetpack' ) + ' ' ) );
+	errorEl.appendChild( spanEl );
```

Comment:
Would this be easier using template strings via backticks and `${}`?

Reply:
I switched to this from a string template as the URL was put there using a string replacement. Concatenation is that not much better. It's still a dangerous surface, even if you try your best escaping the content